src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -594,1024 +594,955 @@ def get_default_realm(self):
                 return realm_info['name']
         return None
 
-    def create_realm(self, realm_name: str, default: bool):
-        rgw_realm_create_cmd = ['realm', 'create']
-        cmd_create_realm_options = ['--rgw-realm', realm_name]
-        if default != 'false':
-            cmd_create_realm_options.append('--default')
-        rgw_realm_create_cmd += cmd_create_realm_options
-        try:
-            exit_code, _, _ = mgr.send_rgwadmin_command(rgw_realm_create_cmd)
-            if exit_code > 0:
-                raise DashboardException(msg='Unable to create realm',
-                                         http_status_code=500, component='rgw')
-        except SubprocessError as error:
-            raise DashboardException(error, http_status_code=500, component='rgw')
+    @RestClient.api_get('/{bucket_name}?versioning')
+    def get_bucket_versioning(self, bucket_name, request=None):
+        """
+        Get bucket versioning.
+        :param str bucket_name: the name of the bucket.
+        :return: versioning info
+        :rtype: Dict
+        """
+        # pylint: disable=unused-argument
+        result = request()
+        if 'Status' not in result:
+            result['Status'] = 'Suspended'
+        if 'MfaDelete' not in result:
+            result['MfaDelete'] = 'Disabled'
+        return result
 
-    def list_realms(self):
-        rgw_realm_list = {}
-        rgw_realm_list_cmd = ['realm', 'list']
-        try:
-            exit_code, out, _ = mgr.send_rgwadmin_command(rgw_realm_list_cmd)
-            if exit_code > 0:
-                raise DashboardException(msg='Unable to fetch realm list',
-                                         http_status_code=500, component='rgw')
-            rgw_realm_list = out
-        except SubprocessError as error:
-            raise DashboardException(error, http_status_code=500, component='rgw')
-        return rgw_realm_list
+    @RestClient.api_put('/{bucket_name}?versioning')
+    def set_bucket_versioning(self, bucket_name, versioning_state, mfa_delete,
+                              mfa_token_serial, mfa_token_pin, request=None):
+        """
+        Set bucket versioning.
+        :param str bucket_name: the name of the bucket.
+        :param str versioning_state:
+            https://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUTVersioningStatus.html
+        :param str mfa_delete: MFA Delete state.
+        :param str mfa_token_serial:
+            https://docs.ceph.com/docs/master/radosgw/mfa/
+        :param str mfa_token_pin: value of a TOTP token at a certain time (auth code)
+        :return: None
+        """
+        # pylint: disable=unused-argument
+        versioning_configuration = ET.Element('VersioningConfiguration')
+        status_element = ET.SubElement(versioning_configuration, 'Status')
+        status_element.text = versioning_state
 
```

Comment:
i am not sure how to review this file? could you explain a bit what you're doing here?

Reply:
Discussed over call